src/server/handlers/thirdpartyRequests/transactions.ts

```diff
@@ -0,0 +1,86 @@
+ use these files except in compliance with the License. You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0 Unless required by applicable law or agreed to in
+ writing, the Mojaloop files are distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS
+ OF ANY KIND, either express or implied. See the License for the specific language governing
+ permissions and limitations under the License. Contributors
+ --------------
+ This is the official list of the Mojaloop project contributors for this file. Names of the original
+ copyright holders (individuals or organizations) should be listed with a '*' in the first column.
+ People who have contributed from an organization can be listed under the organization that actually
+ holds the copyright for their contributions (see the Gates Foundation organization for an example).
+ Those individuals should have their names indented and be marked with a '-'. Email address can be
+ added optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ - Sridhar Voruganti <sridhar.voruganti@modusbox.com>
+
+ --------------
+ ******/
+'use strict'
+
+import { Request, ResponseToolkit, ResponseObject } from '@hapi/hapi'
+import Logger from '@mojaloop/central-services-logger'
+import { ReformatFSPIOPError } from '@mojaloop/central-services-error-handling'
+import { Enum } from '@mojaloop/central-services-shared'
+import { AuditEventAction } from '@mojaloop/event-sdk'
+import Metrics from '@mojaloop/central-services-metrics'
+import { Transactions }  from '~/domain/thirdpartyRequests'
+import { getSpanTags } from '~/shared/util'
+import * as types from '~/interface/types'
+
+/**
+ * summary: CreateThirdpartyTransactionRequests
+ * description: The HTTP request POST /thirdpartyRequests/transactions is used to creation of a transaction request 
+ * for the provided financial transaction in the server.
+ * parameters: body, accept, content-length, content-type, date, x-forwarded-for, fspiop-source,
+ * fspiop-destination, fspiop-encryption,fspiop-signature, fspiop-urifspiop-http-method
+ * produces: application/json 
+ * responses: 202, 400, 401, 403, 404, 405, 406, 501, 503
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const post = async (_context: any, request: Request, h: ResponseToolkit): Promise<ResponseObject> => {
+
+  const histTimerEnd = Metrics.getHistogram(
+    'thirdpartyRequests_transactions_post',
+    'Post thirdpartyRequests transactions request',
+    ['success']
+  ).startTimer()
+  const span = (request as any).span
+
+  try {
+    const tags = getSpanTags(
+      request,
+      Enum.Events.Event.Type.TRANSACTION_REQUEST,
+      Enum.Events.Event.Action.POST)
+    span?.setTags(tags)
+    await span?.audit({
+      headers: request.headers,
+      payload: request.payload
+    }, AuditEventAction.start)
+
+    Transactions.forwardTransactionRequest(
```

Comment:
@eoln did we decide _against_ using `setImmediate` now? I was a fan of it for it's readability.

@sridharvoruganti if we don't do `setImmediate`, then maybe just add a comment above here saying 'we're not awaiting promises' or something

Reply:
@lewisdaly I am also a fan of setImmediate, but there are problems of writing tests using setImmediate: @ramanmangla & @akapur99 have faced problems with jest so I suggest using the asynchronous function without await

Reply:
Ok no problems. We want to make running and testing locally as easy as possible, so let's proceed without `setImmediate`. @sridharvoruganti can you please just add a comment above line 67 with a warning that we are not waiting for a promise here?

Reply:
Ok